### PR TITLE
Add Get Remove Insert and Set waypoints nodes on a PathOfTravel eleme…

### PR DESCRIPTION
**Pull request by ahmedidoumohmed**
…nt. Rework PathOfTravel nodes icons.

### Purpose

As a major user story of Revit Recombinant team (https://jira.autodesk.com/browse/REVIT-146364), the aim of this PR is to create a  set of four new Dynamo nodes to handle and manage WayPoints on a PathOfTravel element. These nodes are named InsertWayPoint, RemoveWayPoint, SetWayPoint and GetWayPoints.
Existing icons of ByFloorPlanPoints, LongestOfShortestExitPaths and Update nodes are slightly modified to keep PathOfTravel icons based on same graphical representations.
image
image
image
image
image

## Description:
The new four nodes are being added to the Dynamo Library under the Revit.Analyze.RouteAnanlysis.PathOfTravel group.
InsertWayPoint, RemoveWayPoint and SetWayPoint nodes return the same updated PathOfTravel element used as input. This allows to chain node operations ob the same PathOfTravel element.
* InsertWayPoint node inserts a WayPoint to PathOfTravel element at the specified index. It takes the pathOfTravel, the wayPoint and the index as input and returns the updated PathOfTravel element.
* RemoveWayPoint node removes the WayPoint at the specified index from PathOfTravel element. It takes the pathOfTravel and the index as input and returns the updated PathOfTravel element.
* SetWayPoint node updates WayPoint at the specified index to the new specified position. It takes the pathOfTravel, the newPosition point and the index as input and returns the updated PathOfTravel element.
* GetWayPoints node returns the WayPoints set from PathOfTravel element. It takes the pathOfTravel as input and returns a list of wayPoints on the PathofTravel.

## Preliminary conditions:
* Input elements (PathOfTravel, Point used to set or to insert) must not be null.
* Indices used to insert, remove or set a WayPoint must be in a valid range.
* For code robustness, an internal validity/integrity check of the input PathOfTravel is performed. If this happens, for some reason, a new warning "Invalid PathOfTravel element." will be shown. There is no explicit case where we can reproduce this particular case.

### Impact assessment

- [ ] Changes the way 3rd party nodes work (Add ikeough to the reviewer list)
- [ ] Includes migrations (Add ikeough to the reviewer list)
- [ ] Adds or changes a public interface (Add lukechurch to the reviewer list)
- [ ] API change is added to the [API Changes document](https://github.com/DynamoDS/Dynamo/wiki/API-Changes).
- [ ] Affects which components of the system depends on which other components (Add pboyer to the reviewer list)
- [x] Introduces new components (Add lukechurch to the reviewer list)
- [ ] Has a performance impact (Add keyu to the reviewer list)
- [ ] Changes the installer (Add sharadkjaiswal to the reviewer list)
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x ] All tests pass using the self-service CI.
- [ x] Snapshot of UI changes, if any.

### Unit tests
The following unit tests have been added to RevitNodesTests assembly and they pass OK (I used RevitTestFrameworkConsole):
WayPointInsertGet_ValidArgs
WayPointSetGet_ValidArgs
WayPointRemoveGet_ValidArgs

### Integration test
New Revit 2021 model is added test\System\PathOfTravel\PathOfTravelsAndWayPoints.rvt with a scenario file GetInsertSetRemoveWayPoints.dyn.
WayPointsGetInsertSetRemove test has been added to RevitSystemTests assembly and it passes OK (I used RevitTestFrameworkConsole).
image
image
image

### Resource localization
New warning exception message "Invalid PathOfTravel element." is added when the PathOfTravel element do not have a valid internal representation.

### Reviewers
@lukechurch 
@awbushnell 
@bykovsm
@aan.du@autodesk.com
@ziyun.shang@autodesk

### FYIs
@awbushnell 
@shafirb
@wangmin

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/PathOfTravel.cs [hunk 1]
```diff
@@ -54,11 +54,11 @@ private PathOfTravel(RvtAnalysis.PathOfTravel rvtPathOfTravel)
       #region Public constructors
 
       /// <summary>
-      /// Calculates the longest Path of Travel of all shortest paths from rooms in the floor plan to given exit points.
+      /// Calculates the longest PathofTravel(s) of all shortest paths from rooms in the floor plan to the specified exit points.
```
Comment (babaivb) [inline src/Libraries/RevitNodes/Elements/PathOfTravel.cs:57]: All changes of "Path of Travel" to "PathofTravel": if the element class name is meant then it should be "PathOfTravel" (capital O).

Comment (ahmedidoumohmed) [reply src/Libraries/RevitNodes/Elements/PathOfTravel.cs]: Changes committed. Thanks for your review.

**Comment by ZiyunShang** [comment]
Hi @ahmedidoumohmed , could you resolve the conflicts? I will merge this PR after the conflicts are resolved.

**Comment by ahmedidoumohmed** [comment]
Hi @ZiyunShang,
the resource files conflicts are resolved.

**Pull request merged**